TickSpec/ServiceProvider.fs

```diff
@@ -3,21 +3,94 @@
 open System
 open System.Collections.Generic
 open System.Diagnostics
+open System.Reflection
 
+/// Provides an instance provider for tests
+type IInstanceProvider =
+    inherit IDisposable
+
+    /// Resolves an instance of desired type
+    abstract member Resolve : Type -> obj
+
+    /// Registers an instance for a type (if there is already a registered instance, it will be replaced)
+    abstract member RegisterInstance : Type -> obj -> unit

```

Comment:
is it a problem for this to be generic (is it much more messy in the codegen) ? Or is it a specific choice from an interface design perspective?

Reply:
I wanted to make the interface in the exact way how it is used by TickSpec. TickSpec will not use the generic variant. So, I tried to make the interface as small as possible. I can easily add generic variants into it.

Reply:
You are correct to model the extensibility point directly as TickSpec wants it and have wrapping elsewhere. (i.e. there's a lot to be said for the IServiceProvider method definition)
My concern is that if its also supposed to be user facing, then there should be a clean idiomatic F# route that allows you to work with type inference in the main